chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -39,11 +133,76 @@ class BraveTextButtonHighlightPathGenerator
 
 namespace views {
 
-// To be called from MdTextButtonBase::UpdateColors().
-void MdTextButtonBase::UpdateColorsForBrave() {
+MdTextButton::MdTextButton(PressedCallback callback,
+                           const std::u16string& text,
+                           int button_context)
+    : MdTextButtonBase(std::move(callback), text, button_context) {
+  SetCornerRadius(100);
+  views::HighlightPathGenerator::Install(
+      this, std::make_unique<BraveTextButtonHighlightPathGenerator>());
+  auto* ink_drop = views::InkDrop::Get(this);
+  views::InkDrop::UseInkDropForFloodFillRipple(ink_drop,
+                                               /*highlight_on_hover=*/false,
+                                               /*highlight_on_focus=*/true);
+  ink_drop->SetCreateHighlightCallback(base::BindRepeating(
+      [](Button* host) {
+        const SkColor fill_color = SK_ColorTRANSPARENT;
+        gfx::RectF boundsF(host->GetLocalBounds());
+        return std::make_unique<InkDropHighlight>(
+            boundsF.size(), static_cast<MdTextButton*>(host)->GetCornerRadius(),
+            boundsF.CenterPoint(), fill_color);
+      },
+      this));
+}
+
+MdTextButton::~MdTextButton() = default;
+
+SkPath MdTextButton::GetHighlightPath() const {
+  SkPath path;
+  int radius = GetCornerRadius();
+  path.addRRect(
+      SkRRect::MakeRectXY(RectToSkRect(GetLocalBounds()), radius, radius));
+  return path;
+}
+
+MdTextButton::Kind MdTextButton::GetKind() const {
+  return kind_;
+}
+
+void MdTextButton::SetKind(Kind kind) {
+  if (kind == kind_)
+    return;
+
+  kind_ = kind;
+
+  // We don't want to affect the OLD style buttons, and we want them to be the
+  // default (for now), so don't change the image-label spacing unless we set
+  // the button kind to something that isn't OLD.
+  if (kind != Kind::kOld) {
+    SetImageLabelSpacing(6);
+  }
+
+  UpdateColors();
+}
+
+void MdTextButton::SetIcon(const gfx::VectorIcon* icon) {
+  icon_ = icon;
+  UpdateColors();
+}
+
+bool MdTextButton::GetLoading() const {
+  return loading_;
+}
+
+void MdTextButton::SetLoading(bool loading) {
+  loading_ = loading;
```

Comment:
Should we call UpdateColors() as loading_ affects color scheme?

Reply:
We sure should! Nice catch!